ethcore/src/ethereum/ethash.rs

```diff
@@ -270,9 +273,14 @@ impl Engine for Arc<Ethash> {
 	/// Apply the block reward on finalisation of the block.
 	/// This assumes that all uncles are valid uncles (i.e. of at least one generation before the current).
 	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<(), Error> {
-		let reward = self.ethash_params.block_reward;
+		let mut reward = self.ethash_params.block_reward;
 		let fields = block.fields_mut();
 
+		let eras = fields.header.number() / self.ethash_params.ecip1017_era_rounds;
+		for _ in 0..eras {
```

Comment:
Does this make `4/5 * reward` also for "Era 0"? 

Winner reward should be full (5 eth) for "Era 0", and 4 eth for Era 1; where "Era 0" == 0-5million[or configured], "Era 1" == 5-10million, etc.

My rust is rusty... maybe should be `1..eras`?